### ISSUE TITLE
RLS between two Authors with "Is Creator" enabled doesn't work for selecting relation.

### ISSUE THREAD

**Issue by maZahaca** [strapi/strapi#18889]
### Required System information

- Environment: development
- OS: darwin-arm64
- Strapi Version: 4.15.4
- Node/Yarn Version: npm/8.19.4 node/v16.20.1 darwin arm64 workspaces/false
- Edition: Community
- Database: postgres
- Is your project Javascript or Typescript: Typescript

### Describe the bug

When enabled "Is Creator" on READ/WRITE/DELETE permissions of the role. I can see only content I've created in the LIST/VIEW/EDIT interface of the entity in my example "Product Category". Which is correct.
But when I go to another Entity EDIT UI, and try to assign "Product Category" to it as a Relation, I'm able to see the whole list of Product Category entities created by other Authors. Which is incorrect.

### Steps to reproduce the behavior

1. Go to strapi admin panel as Super Admin
2. Create 2 data types: Product, and Category which are connected to each other with One-to-many relation.
3. Go to Settings > Roles > Edit Author role, and set the following permissions to both types: CREATE, READ (Is Creator), WRITE (Is Creator), DELETE (Is Creator).
4. Create 2 "Author" users, User 1, and User 2.
5. Log in as a User 1, and create a "Category 1".
6. Log in as a User 2, and start creating "Product 1", as a relation you will be able to see and select "Category 1" created by User 1.

### Expected behavior

I should be able to select only relations which I've created.

### Screenshots

<img width="711" alt="image" src="https://github.com/strapi/strapi/assets/1290067/05368d7e-4a58-411a-b81e-dbc265f0fa73">
<img width="844" alt="image" src="https://github.com/strapi/strapi/assets/1290067/c064e61a-0f16-43b4-a9f1-6db8e55f0dd0">
<img width="1169" alt="image" src="https://github.com/strapi/strapi/assets/1290067/f876ec7b-6098-4206-9192-6ef5fc29b822">

**Comment by afgallo** [comment]
Apologies to Strapi Support Team, I've just opened another bug which seems to be the very same issue @maZahaca has raised, see here: https://github.com/strapi/strapi/issues/18900. Hopefully there is enough details for this to be looked at 🙏🏻

**Comment by maZahaca** [comment]
Yes, @afgallo this sounds the same issue! Hope strapi team have enough time to look it into, as it sounds as a security issue.

**Comment by dcmade01** [comment]
hello how do i create my own website using strapi? ccan someone put me through

**Comment by dcmade01** [comment]
> ## Bug report
> ### Required System information
> * Environment: development
> * OS: darwin-arm64
> * Strapi Version: 4.15.4
> * Node/Yarn Version: npm/8.19.4 node/v16.20.1 darwin arm64 workspaces/false
> * Edition: Community
> * Database: postgres
> * Is your project Javascript or Typescript: Typescript
> 
> ### Describe the bug
> When enabled "Is Creator" on READ/WRITE/DELETE permissions of the role. I can see only content I've created in the LIST/VIEW/EDIT interface of the entity in my example "Product Category". Which is correct. But when I go to another Entity EDIT UI, and try to assign "Product Category" to it as a Relation, I'm able to see the whole list of Product Category entities created by other Authors. Which is incorrect.
> 
> ### Steps to reproduce the behavior
> 1. Go to strapi admin panel as Super Admin
> 2. Create 2 data types: Product, and Category which are connected to each other with One-to-many relation.
> 3. Go to Settings > Roles > Edit Author role, and set the following permissions to both types: CREATE, READ (Is Creator), WRITE (Is Creator), DELETE (Is Creator).
> 4. Create 2 "Author" users, User 1, and User 2.
> 5. Log in as a User 1, and create a "Category 1".
> 6. Log in as a User 2, and start creating "Product 1", as a relation you will be able to see and select "Category 1" created by User 1.
> 
> ### Expected behavior
> I should be able to select only relations which I've created.
> 
> ### Screenshots
> <img alt="image" width="711" src="https://private-user-images.githubusercontent.com/1290067/285195562-05368d7e-4a58-411a-b81e-dbc265f0fa73.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc3NTE0MTUsIm5iZiI6MTcyNzc1MTExNSwicGF0aCI6Ii8xMjkwMDY3LzI4NTE5NTU2Mi0wNTM2OGQ3ZS00YTU4LTQxMWEtYjgxZS1kYmMyNjVmMGZhNzMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMDFUMDI1MTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDI5OTI2NTk3NWMwYTdlODU3YzI5YjU3NDJkODBkZWVkODA2OWMwZjZlNGM0OTU1MWI5MWUxMGVhNWFlNjZlMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.cWXf0vzxHGhTeNrPvK0eLVwMOmE5ezStc0a7lnjxLfc"> <img alt="image" width="844" src="https://private-user-images.githubusercontent.com/1290067/285195726-c064e61a-0f16-43b4-a9f1-6db8e55f0dd0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc3NTE0MTUsIm5iZiI6MTcyNzc1MTExNSwicGF0aCI6Ii8xMjkwMDY3LzI4NTE5NTcyNi1jMDY0ZTYxYS0wZjE2LTQzYjQtYTlmMS02ZGI4ZTU1ZjBkZDAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMDFUMDI1MTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDEzN2U3MGVkMmYyZDNlNDA2Yjc4MWMyNjlkOWU5Y2ZiOTNjMzg5NzI1MGZmNGY3OGI5YWRlOWQxNjUxYTU0YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Kh3z8eriFapgGlbKdCx26dMFHF8HgfApjKk_qB8eKrc"> <img alt="image" width="1169" src="https://private-user-images.githubusercontent.com/1290067/285195929-f876ec7b-6098-4206-9192-6ef5fc29b822.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc3NTE0MTUsIm5iZiI6MTcyNzc1MTExNSwicGF0aCI6Ii8xMjkwMDY3LzI4NTE5NTkyOS1mODc2ZWM3Yi02MDk4LTQyMDYtOTE5Mi02ZWY1ZmMyOWI4MjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMDFUMDI1MTU1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTAzYzk0YzlmN2Y1YWZlYWY5YjYzZDY3NDMxZjZhNmI1NTdhMWFhOWJmMzVhYmQyZTE4ODMxMjRjZDRmNTllNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.3zgz7BQTvLJZe2hq6R6VBrvpwy-dxyQyqk7C8AFrE-s">

does bug work with strapi? im confused
